### PR TITLE
[BE] Admin rights

### PR DESCRIPTION
**Pull request by nspc97**
[BE] Admin rights
https://trello.com/c/5vF8vxMG/87-be-admin-rights

### CODE REVIEW

**Review by MihaiDruta — APPROVED** [review]

File: src/Controller/ActivityController.php [hunk 1]
```diff
@@ -275,7 +275,7 @@ public function getActivityDetails(Activity $activity): Response
         $user = $this->getUser();
         $rights = $this->accessRightsPolicy->canAccessActivity($activity, $user);
 
-        if ($rights === false) {
+        if ($rights === false && $user->getRoles() !== array('ROLE_ADMIN')) {
```
Comment (SpartakusMd) [inline src/Controller/ActivityController.php]: Use `$this->isGranted('ROLE_ADMIN');`

**Review by SpartakusMd — CHANGES_REQUESTED** [review]

File: src/Controller/ActivityController.php (hunk 1 above)
Comment (SpartakusMd) [reply src/Controller/ActivityController.php]: Outside of controllers use `Symfony\Component\Security\Core\Authorization\AuthorizationCheckerInterface::isGranted()`

File: src/Controller/ActivityController.php [hunk 2]
```diff
@@ -685,8 +688,9 @@ public function applyForActivity(
     ): JsonResponse {
         /** @var User $applierUser */
         $applierUser = $this->getUser();
+        $hasAccess = $this->isGranted('ROLE_ADMIN');
 
-        if ($activity->isPublic() === false) {
+        if ($hasAccess && $activity->isPublic() === false) {
```
Comment (SpartakusMd) [inline src/Controller/ActivityController.php]: Should be `if (!$hasAccess`

File: src/Entity/User.php [hunk 3]
```diff
@@ -210,6 +210,12 @@ class User implements UserInterface
      */
     private $stars;
 
+    /**
+     * The role of User
+     * @ORM\Column(type="string")
```
Comment (SpartakusMd) [inline src/Entity/User.php]: You can't save an array as string in database, use `@ORM\Column(type="json")`

File: src/Entity/User.php [hunk 4]
```diff
@@ -367,7 +373,12 @@ public function setUpdatedAt(DateTimeInterface $updatedAt): void
 
     public function getRoles(): array
     {
-        return array('ROLE_USER');
+        return array($this->roles);
+    }
+
+    public function setRoles(string $roles): void
```
Comment (SpartakusMd) [inline src/Entity/User.php]: `$roles` is an array, not string.

**Review by SpartakusMd — CHANGES_REQUESTED** [review]

**Review by SpartakusMd — APPROVED** [review]

**Pull request merged**